net: openthread: radio: Fix platform radio state machine

maciejbaczmanski opened the pull request:
Fix platform radio state machine to be compliant with one shown in OpenThread's `include/openthread/platform/radio.h`

rlubos commented on modules/openthread/platform/diag.c:
```diff
@@ -104,8 +104,8 @@ bool otPlatDiagModeGet(void)
 {
-	ARG_UNUSED(aChannel);
 	sChannel = aChannel;
+	platformRadioChannelSet(aChannel);
```
This doesn't seem related to state machine fixing? IMO the channel setting fix merits a separate commit.

rlubos reviewed (commented):
Looks good, just one remark

maciejbaczmanski committed:
net: openthread: radio: Fix platform radio state machine

Fix platform radio state machine to be compliant with one shown in
OpenThread's `include/openthread/platform/radio.h`

Align twister tests to verify proper behavior of the
state machine.

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

maciejbaczmanski committed:
net: openthread: radio: Add setting channel in diag mode

Add possibility to set channel in diag mode by implementing
`platformRadioChannelSet` API.

Signed-off-by: Maciej Baczmanski <maciej.baczmanski@nordicsemi.no>

edmont commented on modules/openthread/platform/platform-zephyr.h:
```diff
@@ -70,6 +70,16 @@ void platformUartPanic(void);
+/**
+ * Set channel on radio driver.
+ *
+ * @param[in]  aChannel  Channel radio driver should operate on.
```
Just a suggestion.

```suggestion
 * @param[in]  aChannel  The channel that the radio driver should use for operation.
```

maciejbaczmanski replied:
thanks, fixed
